### PR TITLE
Add core data models to the sources build phase

### PR DESCRIPTION
**Pull request by kwridan**
Resolves https://github.com/tuist/tuist/issues/1539

### Short description 📝

- The core data models were previously added to the resources build phase
- Those models need to be included in the sources build phase so they get compiled

### Solution 📦

- Update the build phase generator to include core data models in the sources build phase instead
- Update the fixture to reference some of the core data generated models (e.g. `User`) to help test and validate the correct behaviour

### Implementation 👩‍💻👨‍💻

- [x] Update build phase generator
- [x] Update fixture
- [x] Validate correctness of the change 
  _(should core data models be in Resources or Sources or does it depend on another factor?)_
- [x] Update change log

### Test Plan 🛠

- Run `swift run tuist generate` within `fixtures/ios_app_with_coredata/`
- Open the generated project
- Verify the core data models are within the Sources build phase
- Verify the app builds successfully without any errors

### CODE REVIEW

File: fixtures/ios_app_with_coredata/Sources/Model.swift [hunk 1]
```diff
@@ -0,0 +1,45 @@
+import Foundation
+import CoreData
+
+
+final class MyModel {
```
Comment (pepibumur) [inline fixtures/ios_app_with_coredata/Sources/Model.swift:5]: I wouldn't have thought about adding this test. Great addition!

**Comment by kwridan** [comment]
Digging around further regarding the correctness of the change:

- By default in brand new projects (in Xcode 11) the models are added the sources build phase
- Inspecting the built product we can see `Users.momd` in both cases (when included in resources or in sources)

Screenshot 2020-07-14 at 06 58 41

- By including in sources we get the autogenerated classes working too

I've added an acceptance test to verify the core data files/directories are present in the built product

**Comment by kwridan** [comment]
ouch looks like master has an unstable test 🤐

```
 started.
2867
/Users/runner/work/tuist/tuist/Tests/TuistGeneratorTests/Generator/ProjectFileElementsTests.swift:201: error: -[TuistGeneratorTests.ProjectFileElementsTests test_addElement_xcassets_and_scnassets_multiple_files] : 

XCTAssertEqual failed: 

("Optional(["myfolder/resources/assets.scnassets", "myfolder/resources/assets.xcassets"])")

 is not equal to 

("Optional(["myfolder/resources/assets.xcassets", "myfolder/resources/assets.scnassets"])")
```

**Comment by kwridan** [comment]
it doesn't look like a flake, possibly missed case :/ wonder why CI didn't catch this on the PR

**Pull request merged**